ariadne/cardano/src/Ariadne/Wallet/Cardano/WalletLayer/Types.hs

```diff
@@ -90,4 +90,13 @@ data PassiveWalletLayer m = PassiveWalletLayer
     , pwlLookupKeystore
           :: Kernel.HdRootId
           -> m (Maybe EncryptedSecretKey)
+    , pwlRemoveKey
+          :: [Kernel.HdRootId]
+          -> m ()
+
+    -- | Check keyfile.
+    , pwlGetUnknownKeys
+          :: m [Kernel.HdRootId]
+    , pwlGetWalletsWithoutSecretKeys
+          :: m ([Kernel.WalletName], [Kernel.HdRootId])
```

Comment:
I think it makes more sense to return a list of pairs instead of a pair of lists. It ensures that number of wallet names is the same as the number of root IDs.

Reply:
It will produce only unnecessary zipping/unzipping and it would not ensure us that number of wallet names is the same as the number of root IDs. 
Instead of that, I added checking that for every root ID in the list we have one and only one wallet.